osu.Game/Screens/Edit/Changes/IRevertibleChange.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Screens.Edit.Changes
+{
+    /// <summary>
+    /// Represents a change which can be undone.
+    /// </summary>
+    public interface IRevertibleChange
+    {
+        /// <summary>
+        /// Applies this change to the current state.
+        /// </summary>
+        void Apply();
+
+        /// <summary>
+        /// Applies the inverse of this change to the current state.
+        /// </summary>
+        void Revert();
+    }
+
+    public static class RevertibleChangeExtension
+    {
+        public static void Apply(this IRevertibleChange change, NewBeatmapEditorChangeHandler? changeHandler, bool commitImmediately = false)
```

Comment:
Notice that it's easy not to notice the `Apply()`:
```csharp
new SliderVelocityMultiplierChange(HitObject, proposedVelocity).Apply(changeHandler);
```

Maybe can be:
```csharp
// note: only for the case that NewBeatmapEditorChangeHandler  is nullable.
// use drawable because Drawable support the DI.
public static void ApplyChange(this Drawable drawable, NewBeatmapEditorChangeHandler? changeHandler, IRevertibleChange change, bool commitImmediately = false)
{
}
```

And use like
```csharp
public class HitObjectComposer: CompositeDrawable
{
    [Resolved(canBeNull: true)]
    private NewBeatmapEditorChangeHandler changeHandler { get; set; }

    public void EndPlacement(HitObject hitObject, bool commit)
    {
       ApplyChange(changeHandler, new AddHitObjectChange(EditorBeatmap, hitObject));
    }
}

```

Maye can ask other developer's opinion.

Reply:
Extending `Drawable` for this seems weird.
I think the current `Apply()` is alright. If you forget the `Apply()` altogether you'll get a warning, so you won't miss it. The worst that could happen is you forget to add the `changeHandler` but I think autocomplete will get this right most of the time.